downloads/swan-lake-release-notes/swan-lake-beta4/RELEASE_NOTE.md

```diff
@@ -0,0 +1,830 @@
+
+- The `float:min()` and `float:max()` functions now return `float:NaN` if an argument is `float:NaN`. 
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    float result = float:min(1, float:NaN);
+    io:println(result === float:NaN); // prints true
+
+    result = float:max(5, float:NaN);
+    io:println(result === float:NaN); // prints true
+}
+```
+
+- A bug in the `decimal:fromString()` function which allowed parsing a string that matched the `HexFloatingPointLiteral` has been fixed. It now returns an error.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+   decimal|error result = decimal:fromString("0xab12"); // Now returns an error.
+   io:println(result is error); // prints true
+}
+```
+
+- A bug in the `float:fromString()` function allowed parsing a string that had matched a `DecimalFloatingPointNumber` with `FloatingPointTypeSuffix` has been fixed. This will now return an error.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    float|error result = float:fromString("12.24f"); // Now returns an error
+    io:println(result is error); // prints true
+}
+```
+
+- The `float:fromHexString()` function now returns an error if the provided string argument does not match a `HexFloatingPointLiteral`. 
+
+```ballerina
+import ballerina/io;
+ 
+public function main() {
+   float|error result = float:fromHexString("12.3");
+   io:println(result is error); // prints true, error message - invalid hex string: ‘12.3’
+}
+```
+
+- A spec deviation was fixed in `int:toHexString` which was causing it to convert negative values to a positive number before converting to a hexadecimal string.
+
+- A deviation in the `lang.error` `RetryManager` and `DefaultRetryManager` objects' `shouldRetry` method argument type is fixed. The type has been changed from `error?` to `error`.
+	
+```ballerina
+public class CustomRetryManager {
+   private int count;
+
+   public function init(int count = 3) {
+       self.count = count;
+   }
+
+   public function shouldRetry(error e) returns boolean {
+       if e is CustomError && self.count > 0 {
+           self.count -= 1;
+           return true;
+       } else {
+           return false;
+       }
+   }
+}
+```
+
+- Attempting to use an out of range float value where the applicable contextually-expected type is `float` will now result in a compile-time error.
+
+```ballerina
+public function main() {
+    float x = 945e99876; // Will now result in an error.
+}
+```
+
+- Spec deviations related to identifying the types of numeric literals have been fixed. 
+
+If the numeric literal does not include the float type suffix or the decimal type suffix and if it is not a hex floating-point literal, the type of the numeric literal will be based on the following rules.
+
+1. If the literal is a floating point literal, then the possible basic types in order of preference are `[float, decimal]`; otherwise they are `[int, float, decimal]`.
```

Comment:
```suggestion
1. If the literal is a floating-point literal, then the possible basic types in order of preference are `[float, decimal]`; otherwise they are `[int, float, decimal]`.
```

Reply:
Fixed.